eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/permission/broadcast/PermissionRequestResult.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.permission.broadcast;
+
+import android.app.Activity;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+
+
+/**
+ * The data for the {@link Activity#onRequestPermissionsResult(int, String[], int[])} callback.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class PermissionRequestResult implements Parcelable
+{
+    public final int requestCode;
```

Comment:
I wouldn't call it abstraction layer. Call it proper design. A `PermissionResult` sounds like a well defined entity. From how you use it I'd say that something like 
interface PermissionResult
{
    Iterable<String> granted();
    Iterable<String> denied();
}

Should be almost sufficient. Don't return a list unless you need to access elements by index (i.e. you're actually operating on a list). Returning a list makes no sense here. A set would be more appropriate, but even that doesn't really make sense, because you'd barely check for a specific permission to be in here.

I understand that the interruption of such a request is a rare (error) condition, hence it might not make sense at all to have a separate method to check it. Just throw an appropriate Exception in `granted` and `denied` when called.
Or is there any valid case in which the request can be interrupted (anything but an error)?